src/components/MoneyRequestReportView/MoneyRequestReportTransactionList.tsx

```diff
@@ -181,6 +183,28 @@ function MoneyRequestReportTransactionList({
         [reportActions, sortedTransactions],
     );
 
+    useEffect(() => {
+        const lastFullScreenRoute = navigationRef.getRootState()?.routes.findLast((route) => isFullScreenName(route.name));
+        if (lastFullScreenRoute?.name !== NAVIGATORS.REPORTS_SPLIT_NAVIGATOR && lastFullScreenRoute?.name !== NAVIGATORS.SEARCH_FULLSCREEN_NAVIGATOR) {
+            return;
+        }
+
+        const lastRoute = lastFullScreenRoute?.state?.routes?.at(-1);
+        if (!lastRoute?.params || !('reportID' in lastRoute.params)) {
+            return;
+        }
+        if (lastRoute?.name !== SCREENS.SEARCH.MONEY_REQUEST_REPORT && lastRoute?.name !== SCREENS.REPORT) {
+            return;
+        }
+
+        if (lastRoute.params.reportID !== report.reportID) {
+            return;
+        }
```

Comment:
Some comments here about why we're returning early based on these conditions would be good

Reply:
yes thank you! just requested some changes to conform to our style conventions